compute_sdk/globus_compute_sdk/sdk/client.py

```diff
@@ -587,6 +587,7 @@ def register_function(
             The function to be registered for remote execution
         function_name : str
             The entry point (function name) of the function. Default: None
+            DEPRECATED - function names are taken from the function's __name__ attribute
```

Comment:
As these are nominally [displayed in the documentation](https://globus-compute.readthedocs.io/en/stable/reference/client.html#globus_compute_sdk.Client.register_function), consider making `__name__` monotyped:
```restructuredText
... the function's ``__name__`` ...
```